### PR TITLE
Integrate with CircleCI

### PR DESCRIPTION
**Pull request by ynamiki**
Integrate with CircleCI:

https://circleci.com/gh/nec-blockchain/minbft

### CODE REVIEW

**Review by sergefdrv — COMMENTED** [review]
Would you mind adding a build status badge to the Readme?
```
[![CircleCI](https://circleci.com/gh/nec-blockchain/minbft/tree/master.svg?style=svg)](https://circleci.com/gh/nec-blockchain/minbft/tree/master)
```

**Comment by ynamiki** [comment]
I have added a build status badge to the README.

CircleCI supports [custom Docker images](https://circleci.com/docs/2.0/custom-images/) so it is better to build a custom image with Intel SGX SDK and Go on Ubuntu 16.04 than to do everything in `.circleci/config.yml`. A problem on using a custom image is that we need to have a Docker Registry or to manage an account of Docker Hub. This enhancement is future work.

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+            sudo apt-get update -q
+            sudo apt-get install -qy --no-install-recommends ca-certificates build-essential python wget git make pkg-config
+            wget "https://download.01.org/intel-sgx/linux-2.1.1/ubuntu64-server/sgx_linux_x64_sdk_2.1.101.42529.bin" -O sgx_installer.bin
+            chmod +x sgx_installer.bin
+            echo -e "no\n$(dirname $SGX_SDK)" | sudo ./sgx_installer.bin
+            rm sgx_installer.bin
+            echo '. $SGX_SDK/environment' >> $BASH_ENV
+            echo 'export PATH=$PATH:$SGX_SDK/bin:$SGX_SDK/bin/x64' >> $BASH_ENV
+            echo 'export PKG_CONFIG_PATH=$PKG_CONFIG_PATH:$SGX_SDK/pkgconfig' >> $BASH_ENV
+            echo 'export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$SGX_SDK/sdk_libs' >> $BASH_ENV
```
Comment (sergefdrv) [inline .circleci/config.yml]: `. $SGX_SDK/environment` should update `PATH`, `PKG_CONFIG_PATH`, and `LD_LIBRARY_PATH` environment variables accordingly. Do we really need to duplicate that in lines 28-30?

**Comment by sergefdrv** [comment]
@ynamiki Thanks a lot for taking care of this! This CircleCI integration looks good and should suffice for now. I see how using custom Docker images will make CI more efficient. Let's give it a try later.

**Review by sergefdrv — APPROVED** [review]
The last commit could be squashed with the first one, but never mind, it's not a big deal :slightly_smiling_face:

**Comment by sergefdrv** [comment]
@ynamiki I see "You’re not authorized to merge this pull request." That means I couldn't merge this pull request according to #12

**Comment by ynamiki** [comment]
@sergefdrv Sorry, I'll check settings.

**Comment by ynamiki** [comment]
@sergefdrv I have fixed the permission. Could you try again?

**Pull request merged**